pkg/scheduler/framework/plugins/interpodaffinity/interpod_affinity.go

```diff
@@ -20,47 +20,133 @@ import (
 	"context"
 	"fmt"
 
-	v1 "k8s.io/api/core/v1"
+	"k8s.io/klog"
+
+	"k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/kubernetes/pkg/scheduler/algorithm/predicates"
 	"k8s.io/kubernetes/pkg/scheduler/algorithm/priorities"
 	"k8s.io/kubernetes/pkg/scheduler/framework/plugins/migration"
 	framework "k8s.io/kubernetes/pkg/scheduler/framework/v1alpha1"
+	schedulerlisters "k8s.io/kubernetes/pkg/scheduler/listers"
 	"k8s.io/kubernetes/pkg/scheduler/nodeinfo"
 )
 
 // InterPodAffinity is a plugin that checks inter pod affinity
 type InterPodAffinity struct {
-	handle    framework.FrameworkHandle
-	predicate predicates.FitPredicate
+	snapshotSharedLister schedulerlisters.SharedLister
+	podAffinityChecker   *predicates.PodAffinityChecker
 }
 
+var _ framework.PreFilterPlugin = &InterPodAffinity{}
 var _ framework.FilterPlugin = &InterPodAffinity{}
 var _ framework.ScorePlugin = &InterPodAffinity{}
 
-// Name is the name of the plugin used in the plugin registry and configurations.
-const Name = "InterPodAffinity"
+const (
+	// Name is the name of the plugin used in the plugin registry and configurations.
+	Name = "InterPodAffinity"
+
+	// StateKey is the key in CycleState to InterPodAffinity pre-computed data.
+	// Using the name of the plugin will likely help us avoid collisions with other plugins.
+	preFilterStateKey = "PreFilter" + Name
+)
+
+// preFilterState computed at PreFilter and used at Filter.
+type preFilterState struct {
+	meta *predicates.PodAffinityMetadata
```

Comment:
what about keeping meta as an unnamed member? I assume this object will become the new "meta"

Reply:
it will, but I think it will require casting back and forth, lets keep it as is now, this will naturally get cleaned up when the code is moved.